test/core/color_base/static_transform_gray_to_rgb_fail.cpp

```diff
@@ -0,0 +1,40 @@
+//
+// Copyright 2019 Mateusz Loskot <mateusz at loskot dot net>
+//
+// Distributed under the Boost Software License, Version 1.0
+// See accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt
+//
+#include <boost/gil/color_base_algorithm.hpp>
+#include <boost/gil/pixel.hpp>
+#include <boost/gil/typedefs.hpp>
+
+#include <cassert>
+#include <cstdint>
+
+namespace gil = boost::gil;
+
+int main()
+{
+    // K-element index must be less than size of channel_mapping_t sequence
+    // of the *destination* color base.
+
+    // RGB (wider space) transformation to Gray (narrower space) takes only R channel value
```

Comment:
It isn't immediately obvious to me why this operation should fail. Why doesn't the `static_transform` operation allow arbitrary mappings from source to destination color spaces ?

Reply:
It's how it is implemented as static ir compile-time unrolled iteration through pixel elements (channels) which is also displayed by this assertion(s) 

https://github.com/boostorg/gil/blob/c9ccc8355108b53e06cd62ef88d80cf802bfb5b8/include/boost/gil/color_base_algorithm.hpp#L93-%20L94